ecl2df/wellcompletiondata.py

```diff
@@ -63,7 +71,45 @@ def df(
         wellconnstatus_df = wellconnstatus.df(eclfiles)
         compdat_df = _merge_compdat_and_connstatus(compdat_df, wellconnstatus_df)
 
-    return _aggregate_layer_to_zone(compdat_df)
+    compdat_df = _aggregate_layer_to_zone(compdat_df)
+
+    # Add metadata as an attribute the dataframe
+    meta = _get_metadata(eclfiles)
+    # Slice meta to dataframe columns:
+    compdat_df.attrs["meta"] = {
+        column_key: meta[column_key] for column_key in compdat_df if column_key in meta
+    }
+
+    return compdat_df
+
+
+def _get_ecl_unit_system(eclfiles: EclFiles) -> EclipseUnitSystem:
+    """Returns the unit system of an eclipse deck. The options are \
+    METRIC, FIELD, LAB and PVT-M.
+
+    If none of these are found, the function returns METRIC which is the
+    default unit system in Eclipse.
+    """
+    unit_systems = [unitsystem.value for unitsystem in EclipseUnitSystem]
+    for keyword in eclfiles.get_ecldeck():
+        if keyword.name in unit_systems:
+            return EclipseUnitSystem(keyword.name)
+    return EclipseUnitSystem.METRIC
+
+
+def _get_metadata(eclfiles: EclFiles) -> Dict[str, Dict[str, Any]]:
+    """Provide metadata for the well completion data export"""
+    meta: Dict[str, Dict[str, Any]] = {}
+    unitsystem = _get_ecl_unit_system(eclfiles)
+    kh_units = {
+        EclipseUnitSystem.METRIC: "mDm",
+        EclipseUnitSystem.FIELD: "mDft",
+        EclipseUnitSystem.LAB: "mDcm",
+        EclipseUnitSystem.PVTM: "mDm",
+    }
```

Comment:
Should this also be an `Enum` at the top of the file?

Reply:
Do you mean like this. I still need to keep the mapping somewhere, or do you have another idea?

```
    meta: Dict[str, Dict[str, Any]] = {}
    unitsystem = _get_ecl_unit_system(eclfiles)
    kh_units = {
        EclipseUnitSystem.METRIC: KHUnit.METRIC,
        EclipseUnitSystem.FIELD: KHUnit.FIELD,
        EclipseUnitSystem.LAB: KHUnit.LAB,
        EclipseUnitSystem.PVTM: KHUnit.PVTM,
    }
    meta["KH"] = {}
    meta["KH"]["unit"] = kh_units[unitsystem].value
    return meta
```